runtime-light/tl/tl-types.h

```diff
@@ -4,16 +4,237 @@
 
 #pragma once
 
+#include <algorithm>
+#include <cstddef>
 #include <cstdint>
+#include <optional>
 
+#include "common/tl/constants/common.h"
+#include "runtime-core/allocator/runtime-allocator.h"
+#include "runtime-core/memory-resource/resource_allocator.h"
+#include "runtime-core/memory-resource/unsynchronized_pool_resource.h"
 #include "runtime-core/runtime-core.h"
 #include "runtime-light/tl/tl-core.h"
 
 namespace tl {
 
-// ===== JOB WORKERS =====
+// ===== GENERAL =====
+
+struct Bool final {
+  bool value{};
+
+  bool fetch(TLBuffer &tlb) noexcept {
+    const auto magic{tlb.fetch_trivial<uint32_t>().value_or(TL_ZERO)};
+    value = magic == TL_BOOL_TRUE;
+    return magic == TL_BOOL_TRUE || magic == TL_BOOL_FALSE;
+  }
+
+  void store(TLBuffer &tlb) const noexcept {
+    tlb.store_trivial<uint32_t>(value ? TL_BOOL_TRUE : TL_BOOL_FALSE);
+  }
+};
+
+template<tl_serializable T>
+struct Maybe final {
+  std::optional<T> opt_value{};
+
+  bool fetch(TLBuffer &tlb) noexcept {
+    const auto magic{tlb.fetch_trivial<uint32_t>().value_or(TL_ZERO)};
+    if (magic == TL_MAYBE_TRUE) {
+      opt_value.emplace();
+      return (*opt_value).fetch(tlb);
+    } else if (magic == TL_MAYBE_FALSE) {
+      opt_value = std::nullopt;
+      return true;
+    }
+    return false;
+  }
+
+  void store(TLBuffer &tlb) const noexcept {
+    if (opt_value.has_value()) {
+      tlb.store_trivial<uint32_t>(TL_MAYBE_TRUE);
+      (*opt_value).store(tlb);
+    } else {
+      tlb.store_trivial<uint32_t>(TL_MAYBE_FALSE);
+    }
+  }
+};
+
+template<tl_serializable T>
```

Comment:
I think it's better to distinguish `tl_fetchable` and `tl_storable` and move such concepts to fetch/store methods.
For example, I have a structure that are stored by KPHP, but read from k2. And I want to add new TL-function that will work with vector of such a structure. Then I'll have to implement fetch method, that will never be called in KPHP. It leads  dead code or forces to write stubs that are bad things

Reply:
Nice point. Fixed